### PR TITLE
Fix sptagindexer

### PR DESCRIPTION
**Pull request by JoanFM**
This is one step to fix the docker build, but test is still failing because the code does not do what it should

### CODE REVIEW

File: indexers/vector/SptagIndexer/__init__.py [hunk 1]
```diff
@@ -44,11 +44,15 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
         _index.SetBuildParam("NumberOfThreads", str(self.num_threads))
         _index.SetBuildParam("DistCalcMethod", self.space)
 
-        if _index.Build(vecs, vecs.shape[0]):
+        if _index.Build(vecs.astype('float32'), vecs.shape[0]):
```
Comment (JoanFM) [inline indexers/vector/SptagIndexer/__init__.py:47]: In their example, they Build, save to a new directory and load from the directory? Is it needed here? And also they have a different mode where they use a client to a remote indexer

File: indexers/vector/NmsLibIndexer/Dockerfile [hunk 2]
```diff
@@ -4,6 +4,7 @@ FROM jinaai/jina
 COPY . /workspace
 WORKDIR /workspace
 
+RUN apt-get update && apt-get install --no-install-recommends -y gcc libboost-all-dev libgsl0-dev libeigen3-dev
```
Comment (JoanFM) [inline indexers/vector/NmsLibIndexer/Dockerfile:7]: Remove this as not part of the PR?

File: indexers/vector/SptagIndexer/__init__.py (hunk 1 above)
Comment (nan-wang) [reply indexers/vector/SptagIndexer/__init__.py:47]: I'm following this tutorial, https://github.com/microsoft/SPTAG/blob/master/docs/Tutorial.ipynb

Comment (JoanFM) [reply indexers/vector/SptagIndexer/__init__.py:47]: Yes, this is the one I am talking about, but maybe the `Save` is just for persistence and not necessary. If the tests pass it is good

Comment (nan-wang) [reply indexers/vector/SptagIndexer/__init__.py:47]: Yes, `Save` is  for saving the index for later usages. It is not required by the `Search`.
